src/prelude/string.mare

```diff
@@ -1,3 +1,4 @@
+:import "collections" (Count)
```

Comment:
I think I want to refactor this to not use `Count` - just because I don't want the `prelude` to have a dependency on any other package, including the `collections` package.

But I can do that refactor myself after merging.

Thanks for all of this!

Reply:
It turns out that having this dependency is causing other parts of the test suite in the CI run to break.

So I'm going to commit my fix to this branch, simultaneously refactoring `String.substring` to use the more performant unsafe CPointer operations.